src/unistd.rs

```diff
@@ -3992,3 +3992,58 @@ pub fn chflags<P: ?Sized + NixPath>(path: &P, flags: FileFlag) -> Result<()> {
     Errno::result(res).map(drop)
 }
 }
+
+#[cfg(any(
+    all(target_os = "linux", target_env = "gnu"),
+    target_os = "freebsd"
+))]
+#[cfg(feature = "fs")]
+libc_bitflags! {
+    /// Options for close_range()
+    #[cfg_attr(docsrs, doc(cfg(feature = "fs")))]
+    pub struct CloseRangeFlags : c_uint {
+        #[cfg(all(target_os = "linux", target_env = "gnu"))]
+        /// Unshare the file descriptors range before closing them
+        CLOSE_RANGE_UNSHARE;
+        /// Set the close-on-exec flag on the file descriptors range
+        CLOSE_RANGE_CLOEXEC;
+    }
+}
+
+feature! {
+#![feature = "fs"]
+
+/// Close all the file descriptor from a given range.
+/// An optional flag can be applied to modify its behavior.
+#[cfg(any(
+    all(target_os = "linux", target_env = "gnu"),
+    target_os = "freebsd"
+))]
+pub fn close_range<F: std::os::fd::AsFd>(fdbegin: F, fdlast: F, flags: CloseRangeFlags) -> Result<Option<c_int>> {
+    use std::os::fd::AsRawFd;
+
+    let raw = unsafe {
+        Errno::clear();
+
+        cfg_if! {
+            if #[cfg(all(target_os = "linux", target_env = "gnu"))] {
+                libc::syscall(libc::SYS_close_range, fdbegin.as_fd().as_raw_fd() as u32, fdlast.as_fd().as_raw_fd() as u32, flags.bits() as i32)
```

Comment:
glibc added the wrapper in 2.34, and the libc crate [has it exposed](https://docs.rs/libc/latest/libc/fn.close_range.html), any reason why we are using a raw syscall here?

Reply:
Originally I call the wrapper but I think it is due to the CI when there is old glibc

Reply:
Makes sense, some CIs are still using Ubuntu 20.04 due to [this issue](https://github.com/nix-rust/nix/issues/2418), which has glibc 2.31 😮‍💨, perhaps we can try bumping them to see if we are lucky now

Reply:
we can do that if we do not need to bother supporting older releases out in the field. I may come back to it later.

Reply:
Yeah, if we use glibc wrapper, then using this interface would require glibc 2.34 or above. We are still not lucky regarding that test: #2533, perhaps we should disable the test under QEMU then 😪